_lzbench/lzbench.cpp

```diff
@@ -233,6 +233,20 @@ size_t common(uint8_t *p1, uint8_t *p2)
     return size;
 }
 
+/*
+ * Allocate a buffer of size bytes using malloc (or equivalent call returning a buffer
+ * that can be passed to free). Touches each page so that the each page is actually
+ * physically allocated and mapped into the process.
+ */
+void *alloc_and_touch(size_t size, bool must_zero) {
```

Comment:
This is the function that does the `malloc` or `calloc` and then touches each page. You might imagine that you could just do a `memset(buf, 0, size)` instead, but in fact tricky compilers like `gcc` can recognize this `malloc + memset` idiom, place it with a `calloc` and then we get the same behavior as before.